### PR TITLE
Handle nested column types properly for empty parquet files.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Fixes:  https://github.com/rapidsai/cudf/issues/8323

Also fixes a recently introduced bug in the test column equality checker.  The code was previously relying on accesses to device memory being transparently handled by `thrust::device_vector`

### CODE REVIEW

File: cpp/src/io/parquet/reader_impl.cu [hunk 1]
```diff
@@ -37,6 +37,8 @@
 #include <numeric>
 #include <regex>
 
+#include <cudf_test/column_utilities.hpp>
```
Comment (devavret) [inline cpp/src/io/parquet/reader_impl.cu:40]: Why is a cudf test header included in a library source?

Comment (nvdbaranec) [reply cpp/src/io/parquet/reader_impl.cu:40]: Good catch.  Lingering debug code.

**Review by hyperbolic2346 — APPROVED** [review]

**Review by devavret — APPROVED** [review]
Looks fine. A bit of code repetition but this seems more urgent.

**Review by ttnghia — APPROVED** [review]

**Comment by nvdbaranec** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by nvdbaranec** [comment]
rerun tests

**Comment by ttnghia** [comment]
Rerun tests.

**Pull request merged**